src/main/java/robot/subsystems/turret/Turret.java

```diff
@@ -0,0 +1,149 @@
+package robot.subsystems.turret;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.LimitSwitchNormal;
+import com.ctre.phoenix.motorcontrol.LimitSwitchSource;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import robot.subsystems.turret.commands.TurnTurret;
+
+import static robot.Constants.Turret.*;
+import static robot.Ports.Turret.MASTER;
+
+/**
+ * @author Adam & Barel
+ * @version 1.0
+ * <p>
+ * this class includes methods for the Turret.
+ * {@using TalonSRX}
+ * {@using Relative Encoder}
+ * {@using Hall Effect}
+ */
+public class Turret extends Subsystem {
+    private TalonSRX master = new TalonSRX(MASTER);
+    private double angle;
+    private Direction direction;
+
+    /**
+     * configures the encoder and PID constants.
+     */
+    public Turret() {
+        direction = Direction.RIGHT;
+        master.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, TIMEOUT_MS);
+        master.config_kP(TALON_PID_SLOT, KP, TIMEOUT_MS);
+        master.config_kI(TALON_PID_SLOT, KI, TIMEOUT_MS);
+        master.config_kD(TALON_PID_SLOT, KD, TIMEOUT_MS);
+        master.config_kF(TALON_PID_SLOT, KF, TIMEOUT_MS);
+        master.setInverted(IS_MASTER_INVERTED);
+        master.configPeakCurrentLimit(MAX_CURRENT);
+        master.configForwardLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen);
+        reset();
+        master.setSelectedSensorPosition(0);
+    }
+
+    @Override
+    public void initDefaultCommand() {
+        setDefaultCommand(new TurnTurret(45));

```

Comment:
I would just leave the turret at its last position. If you shoot balls and then going somewhere else then returning to shoot, it is probable the wanted angle would not change significantly.